realm/realm-library/src/androidTest/java/io/realm/RealmResultsTests.java

```diff
@@ -1440,6 +1450,211 @@ public void distinct_invalidTypesLinkedFields() {
         }
     }
 
+    // distinctAsync
+    private Realm openRealmInstance(String name) {
+        RealmConfiguration config = configFactory.createConfiguration(name);
+        Realm.deleteRealm(config);
+        return Realm.getInstance(config);
+    }
+
+    private void populateTestRealm(Realm testRealm, int objects) {
+        testRealm.beginTransaction();
+        testRealm.allObjects(AllTypes.class).clear();
+        testRealm.allObjects(NonLatinFieldNames.class).clear();
+        for (int i = 0; i < objects; ++i) {
+            AllTypes allTypes = testRealm.createObject(AllTypes.class);
+            allTypes.setColumnBoolean((i % 3) == 0);
+            allTypes.setColumnBinary(new byte[]{1, 2, 3});
+            allTypes.setColumnDate(new Date(DECADE_MILLIS * (i - (objects / 2))));
+            allTypes.setColumnDouble(3.1415);
+            allTypes.setColumnFloat(1.234567f + i);
+            allTypes.setColumnString("test data " + i);
+            allTypes.setColumnLong(i);
+            NonLatinFieldNames nonLatinFieldNames = testRealm.createObject(NonLatinFieldNames.class);
+            nonLatinFieldNames.set델타(i);
+            nonLatinFieldNames.setΔέλτα(i);
+            nonLatinFieldNames.set베타(1.234567f + i);
+            nonLatinFieldNames.setΒήτα(1.234567f + i);
+        }
+        testRealm.commitTransaction();
+    }
+
+    @Test
+    @RunTestInLooperThread
+    public void distinctAsync() throws Throwable {
+        final AtomicInteger changeListenerCalled = new AtomicInteger(4);
+        final Realm realm = looperThread.realm;
+        final long numberOfBlocks = 25;
+        final long numberOfObjects = 10; // must be greater than 1
+        populateForDistinct(realm, numberOfBlocks, numberOfObjects, false);
+
+        final RealmResults<AnnotationIndexTypes> distinctBool = realm.where(AnnotationIndexTypes.class).findAll().distinctAsync("indexBoolean");
```

Comment:
Use `AnnotationIndexTypes.FIELD_INDEXEDBOOLEAN` or similar. All fields should ideally be referenced through constants.